api/src/modules/sections/sections.service.ts

```diff
@@ -27,6 +27,26 @@ export class SectionsService extends AppBaseService<
     super(sectionsRepository, 'section', 'sections');
   }
 
+  public async searchSectionsWithData(
+    query: FetchSpecification & WidgetDataFiltersSchema,
+  ): Promise<SectionWithDataWidget[]> {
+    // TODO: There is a bug / weird behavior in typeorm when using take and skip with leftJoinAndSelect:
+    //       https://github.com/typeorm/typeorm/issues/4742#issuecomment-780702477
+    //       Since we don't need pagination for this endpoint, we can disable it for now but worth checking
+    query.disablePagination = true;
+    const [sections] = await this.findAll(query);
+
+    const { filters } = query;
+
+    const sectionsWithData =
+      await this.surveyDataRepository.addSurveyDataToSections(
+        sections as Section[],
```

Comment:
Why is this parsing needed?  `findAll` should return Section[]

Reply:
Typorm returns Partial<Section[]> and  this.surveyDataRepository.addSurveyDataToSection expects Section[]

Reply:
I guess it expects it because the method signature defines it as is, however it can be accurate to receive a Partial section depending on how the consumer filters it.

This being said, I think in this context is not reallistic that the Section will be retrieved partially